src/OptionsCompounderAave2.sol

```diff
@@ -22,92 +26,56 @@ error OptionsCompounder__FlashloanNotProfitable(
-            revert OptionsCompounder__NotAStrategy();
-        }
-        _;
-    }
+    ILendingPool private lendingPool;
+    address wantToken;
+    bool flashloanFinished;
+    uint256 gain = 0; // TODO: remove at the end
 
     /**
      * List of params which are initiated at the begining:
      * @param _optionToken - option token address which allows to redeem underlying token via operation "exercise"
      * @param _addressProvider - address lending pool address provider - necessary for flashloan operations
      * @param _reaperSwapper - address to contract allowing to swap tokens in easy way
-     * @param _strategies - strategies which can use this module
      * */
-    constructor(
+    // TODO: Add onlyInitializing !
+    function _OptionsCompounder_Init(
         address _optionToken,
         address _addressProvider,
         address _reaperSwapper,
-        address[] memory _strategies
-    )
-        FlashLoanReceiverBase(ILendingPoolAddressesProvider(_addressProvider))
-        Ownable(msg.sender)
-    {
+        address _wantToken
+    ) internal /* onlyInitializing */ {
         optionToken = IOptionsToken(_optionToken);
+        addressProvider = ILendingPoolAddressesProvider(_addressProvider);
+        lendingPool = ILendingPool(addressProvider.getLendingPool());
         swapperSwaps = ISwapperSwaps(_reaperSwapper);
-
-        /* address to balancer-like vault (default is address to Beetx vault on optimism chain) */
-        vaultAddress = BEETX_VAULT_OP;
-        for (uint8 idx = 0; idx < _strategies.length; idx++) {
-            strategies.push(_strategies[idx]);
-        }
+        wantToken = _wantToken;
+        flashloanFinished = true;
     }
 
     /***************************** Setters ***********************************/
-    /* Only owner functions */
-    function addStrategy(address _strategy) external onlyOwner {
-        if (false == isStrategyAdded(_strategy)) {
-            strategies.push(_strategy);
-        } else {
-            revert OptionsCompounder__StrategyAlreadyExists();
-        }
-    }
-
-    function removeStrategy(address _strategy) external onlyOwner {
-        address[] memory tmpStrategies = strategies;
-        bool strategyFound = false;
-        for (uint8 idx = 0; idx < tmpStrategies.length; idx++) {
-            if (
-                (tmpStrategies[idx] == _strategy) &&
-                (idx != (tmpStrategies.length - 1))
-            ) {
-                strategies[idx] = tmpStrategies[tmpStrategies.length - 1];
-                strategyFound = true;
-            }
-        }
-        if (false != strategyFound) {
-            strategies.pop();
-        } else {
-            revert OptionsCompounder__StrategyNotFound();
-        }
-    }
-
-    function setSwapper(address _swapper) external onlyOwner {
+    /* Only owner functions - in the future multi level access control*/
+    /* TODO: Access control to add ! */
+    function setSwapper(address _swapper) external {
```

Comment:
strategy itself has this functionality, we should not make this contract have a separate setter and variable than its child for the swapper